### PR TITLE
Fix OpenTracing shim under legacy AspNetCore activities

### PR DESCRIPTION
**Pull request by pjanotti**
## Changes

Fixes OpenTracing spans under the legacy Activity "Microsoft.AspNetCore.Hosting.HttpRequestIn" (enabled by AspNetCore instrumentation package). The OpenTracing spans under the legacy Activity can be missing, having broken parent/active span relationships and cause the premature termination of the legacy activity.

The special handling for "Microsoft.AspNetCore.Hosting.HttpRequestIn" on the OpenTracing shim was present since the first version of the shim, see #197. It got its current behavior on #994. However, that behavior makes the first OT span under the legacy activity to not be created, the legacy activity to be stopped where the first OT span finishes, and any subsequent OT span to be created under the context of the legacy activity parent (which typically means creating a new trace).

The OpenTracing spans should be integrated with OpenTelemetry traces, consider the case of third-party libraries instrumented with OpenTracing. The fix simply consists in removing the special case for "Microsoft.AspNetCore.Hosting.HttpRequestIn".

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes

### CODE REVIEW

**Review by Kielek — APPROVED** [review]

**Comment by pjanotti** [comment]
@open-telemetry/dotnet-approvers I'm planning to join the SIG meeting today in case you have any Q about this PR.

**Comment by rwkarg** [comment]
> @open-telemetry/dotnet-approvers I'm planning to join the SIG meeting today in case you have any Q about this PR.

Can't make the SIG, but we're running a local fork with this PR and it's resolved the broken trace issues we previously had when using the shim.

**Review by alanwest — APPROVED** [review]
Approving since this PR seems to take things in the right direction. It might even resolve #2257.

Though there's a lot about this shim that seems kinda rough. For starters, it wraps the OpenTelemetry shim API which in turn wraps the Activity API. Probably would make more sense to use the Activity API directly.

File: test/OpenTelemetry.Shims.OpenTracing.Tests/SpanBuilderShimTests.cs [hunk 1]
```diff
@@ -330,5 +330,29 @@ public void Start()
             Assert.NotNull(span);
             Assert.Equal("foo", span.Span.Activity.OperationName);
         }
+
+        [Fact]
+        public void Start_UnderAspNetCoreInstrumentation()
```
[the remark this answers is not in the document]
Comment (pjanotti) [reply test/OpenTelemetry.Shims.OpenTracing.Tests/SpanBuilderShimTests.cs:335]: The old code fails on the assert at line 350

[the remark this answers is not in the document]
Comment (pjanotti) [reply test/OpenTelemetry.Shims.OpenTracing.Tests/SpanBuilderShimTests.cs:335]: also on 355 if check of 350 was removed.

File: test/OpenTelemetry.Shims.OpenTracing.Tests/SpanBuilderShimTests.cs [hunk 2]
```diff
@@ -127,13 +127,13 @@ public void Start_ActivityOperationRootSpanChecks()
 
             // matching root operation name
             var tracer = TracerProvider.Default.GetTracer(TracerName);
-            var shim = new SpanBuilderShim(tracer, "foo", new List<string> { "foo" });
+            var shim = new SpanBuilderShim(tracer, "foo");
```
[the remark this answers is not in the document]
Comment (pjanotti) [reply test/OpenTelemetry.Shims.OpenTracing.Tests/SpanBuilderShimTests.cs:130]: yeah, it was allocating on every shim for nothing.

**Comment by pjanotti** [comment]
@alanwest 
> Though there's a lot about this shim that seems kinda rough. For starters, it wraps the OpenTelemetry shim API which in turn wraps the Activity API. Probably would make more sense to use the Activity API directly.

Yeah, I think that reviewing the whole shim seems a good idea.

**Comment by pjanotti** [comment]
> It might even resolve #2257 

Yes, this PR should fix #2257 and #2787 too.

**Pull request merged**